train.c

```diff
@@ -0,0 +1,90 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <float.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#include "agents/temporal_difference.h"
+#include "game.h"
+
+static double epsilon = 0.3;
+#define NUM_ITERATION 10000
+#define RAND_UNIFORM ((float) rand() / (float) RAND_MAX)
+static unsigned int N_STATES = 1;
+static td_agent_t agent[2];
+
+int get_action_epsilon_greedy(char *table, td_agent_t *agent)
+{
+    int move_cnt = 0;
+    int *available_moves = get_available_moves(table, &move_cnt);
+    if (RAND_UNIFORM < epsilon) {  // explore
+        printf("explore %d\n", available_moves[rand() % move_cnt]);
+        return available_moves[rand() % move_cnt];
+    }
+    int act = get_action_exploit(table, agent);
+    epsilon *= GAMMA;
```

Comment:
It seems that you are decaying `epsilon` by `GAMMA`, however, `GAMMA` is the parameter of the "Markov Decision Model". Does it have other meanings, or you should use another parameter like `decay_rate` to replace `GAMMA` here.